### PR TITLE
Use new selectmenu anatomy in explainer

### PR DESCRIPTION
**Pull request by josepharhar**
This new anatomy was proposed and agreed on here:
https://github.com/openui/open-ui/issues/702

### CODE REVIEW

**Comment by josepharhar** [comment]
@mfreed7 want to review this?

File: site/src/pages/components/selectlist.mdx
Comment (gregwhitworth) [inline site/src/pages/components/selectlist.mdx:49]: @josepharhar I propose not using the term slot anywhere and this anatomy should not have any slots and thus not have the "toggle".

Comment (gregwhitworth) [reply site/src/pages/components/selectlist.mdx:49]: Additionally, I don't get this statement: 

> The portion of the element which is always visible and in flow with the page, unlike the listbox.

It may not be visible and it may not be in flow (eg: I should be able to position this out of flow). I think a more clear definition would be that this will be the element that invokes the listbox as a popover via the popover API or something to that affect

File: site/src/pages/components/selectlist.mdx
Comment (gregwhitworth) [inline site/src/pages/components/selectlist.mdx:52]: Wrap selectlist with code quotes

File: site/src/pages/components/selectlist.mdx
Comment (gregwhitworth) [inline site/src/pages/components/selectlist.mdx:65]: Are these psuedo elements only available if the author doesn't provide their own element? At which point I should be able to simply style them as elements, correct?

File: site/src/pages/components/selectlist.mdx
Comment (gregwhitworth) [inline site/src/pages/components/selectlist.mdx:68]: This is no longer the case and should be removed

**Review by gregwhitworth — CHANGES_REQUESTED** [review]
Following signoff from @mfreed7 can we also get @domenic and possibly Anne to give it a review?

File: site/src/components/select-anatomy.jsx
Comment (lukewarlow) [inline site/src/components/select-anatomy.jsx:10]: Should the part name here be selectedvalue or just value?

Reason I ask is virtually all inputs will have some form of "value" part. But `selectedvalue` only really makes sense for elements with multiple choices? Saying that most elements can be linked up to a datalist so maybe `selectedvalue` is fine for all of them?

File: site/src/pages/components/selectlist.mdx
Comment (josepharhar) [reply site/src/pages/components/selectlist.mdx:49]: > I propose not using the term slot anywhere and this anatomy should not have any slots and thus not have the "toggle".

When I talked to domenic earlier, he didn't sound very opposed to defining the user-agent shadowroot of selectlist in the HTML spec because we already sort of do so for the details element and mention "slots": https://html.spec.whatwg.org/multipage/rendering.html#the-details-and-summary-elements

For this reason, I feel comfortable going forward using this terminology until we encounter further issues. Is there any other reason you don't want to use the word slot?

> this will be the element that invokes the listbox as a popover

But this is exactly what `<button type=selectmenu>` is. I could get rid of the "button (slot)" section entirely, but I think its worth spelling out since we are explicitly choosing to have a separate button slot and button type=selectmenu in order to support split buttons

> It may not be visible and it may not be in flow (eg: I should be able to position this out of flow).

Good point, I rewrote the first sentence

File: site/src/pages/components/selectlist.mdx
Comment (josepharhar) [reply site/src/pages/components/selectlist.mdx:52]: done for the highlighted section

File: site/src/pages/components/selectlist.mdx
Comment (josepharhar) [reply site/src/pages/components/selectlist.mdx:65]: They should target whatever is rendered, whether thats the default/fallback content or the author supplied one. I added a sentence to clarify

File: site/src/pages/components/selectlist.mdx
Comment (josepharhar) [reply site/src/pages/components/selectlist.mdx:68]: whoops, good catch. i removed this sentence

File: site/src/components/select-anatomy.jsx
Comment (josepharhar) [reply site/src/components/select-anatomy.jsx:10]: The new `<selectedvalue>` element is only intended to work for `<selectlist>`, at least for now, and it is an element with multiple choices.

Do you think the element's tag name should be `<value>` instead?

**Comment by josepharhar** [comment]
> can we also get @domenic and possibly Anne to give it a review?

imo they should review it on the public site after this and my remaining patches are merged to the explainer

File: site/src/components/select-anatomy.jsx
Comment (lukewarlow) [reply site/src/components/select-anatomy.jsx:10]: If it were to be value then it likely could be reused for other inputs in future?

Comment (josepharhar) [reply site/src/components/select-anatomy.jsx:10]: Good point. I opened an issue to discuss this further: https://github.com/openui/open-ui/issues/808

File: site/src/pages/components/selectlist.mdx
Comment (mfreed7) [reply site/src/pages/components/selectlist.mdx]: +1 we should mention how this works via slots. @domenic ’s point was about the developer side of this - that you shouldn’t need to use the `slot` attribute. But I think it’s important (for interop, spec ease, and stylability) that we define the structure of the shadow root, and define the algorithm used to slot in content.

**Comment by mfreed7** [comment]
> > can we also get @domenic and possibly Anne to give it a review?
> 
> imo they should review it on the public site after this and my remaining patches are merged to the explainer

+100 I really don’t like that we have so much review happening on these explainers. The feedback is awesome, and it’s not that I don’t want all of the great input! But the history and conversation on a PR is so much harder to follow later, as compared to an issue. I’d really prefer we do a very basic review on these and then land them. Fundamental (or even small) GitHub issues should then be raised to discuss changes, and the explainer gets updated after we resolve those issues. That way we have a paper trail for how we got to where we ended up.

**Review by mfreed7 — APPROVED** [review]

**Review by gregwhitworth — APPROVED** [review]

**Pull request merged**
